### ISSUE TITLE
Cluster stays if tapped at map borders (Android/iOS)

### ISSUE THREAD

**Issue by TomRiedl** [Leaflet/Leaflet.markercluster#529]
The correct behavior:
![working1](https://cloud.githubusercontent.com/assets/8522918/8878857/dbf9f3cc-322d-11e5-87c9-d56d3c3b1efa.png)
![working2](https://cloud.githubusercontent.com/assets/8522918/8878860/dfbe65e2-322d-11e5-8a0f-0f3051f9bcc6.png)
This works fine if the cluster is near the center of the map.

But if I move the map to a position where the cluster is near the border of the map, this happens:
![notworking1](https://cloud.githubusercontent.com/assets/8522918/8878891/06bb685c-322e-11e5-8f04-fd8c355aee91.png)
![notworking2](https://cloud.githubusercontent.com/assets/8522918/8878893/087a6968-322e-11e5-98d1-cb8aa6a6d9fe.png)

This is reproducible every time on different mobile devices (tested on two Android and three iOS devices).
This is not reproducible in Firefox and Chrome.

The app is build through PhoneGap.

The code the clusters are created:

``` javascript
this._map = L.map('mapNear', { worldCopyJump: true, });
this._map.options.minZoom = 1;
L.control.scale().addTo(this._map);

L.tileLayer(Application.AppCore.Configuration.MapTiles, {
    attribution: '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
}).addTo(this._map);


this._markerCluster = new L.MarkerClusterGroup({ showCoverageOnHover: false });
for (index in items)
{
    [...]
    marker = L.marker([item.Coordinates.Latitude, item.Coordinates.Longitude],
        {
            icon: L.AwesomeMarkers.icon({ icon: item.Icon ? item.Icon : 'cube', markerColor: 'blue', prefix: 'fa', spin: false }),
            zIndexOffset: 10000
        });
    marker.bindPopup('<some html code>');
    this._markerCluster.addLayer(marker);
}
this._markerCluster.addTo(this._map);
```

**Comment by danzel** [comment]
https://github.com/Leaflet/Leaflet.markercluster/blob/master/src/MarkerClusterGroup.js#L892-L893
If you change this so instead of 0 it is a small number does that fix it?
Or make it always 0 and it may reproduce on desktop.

**Comment by ThomasRiedl** [comment]
The changes work like you expected it to happen:

With
latDiff = 0,
lngDiff = 0;
-> reproducible also on desktop.

With
latDiff = Math.abs(sw.lat - ne.lat),
lngDiff = Math.abs(sw.lng - ne.lng);
-> No bugs on mobile devices

**Comment by ftechiesnitin** [comment]
Hi @ThomasRiedl, I am facing the same issue for one of my app. Can you tell me how you fixed this issue?

Thanks.

**Comment by ThomasRiedl** [comment]
Hi @ftechiesnitin, I replaced
`latDiff = L.Browser.mobile ? 0 : Math.abs(sw.lat - ne.lat),`
`lngDiff = L.Browser.mobile ? 0 : Math.abs(sw.lng - ne.lng);`

with

`latDiff = Math.abs(sw.lat - ne.lat),`
`lngDiff = Math.abs(sw.lng - ne.lng);`

in file MarkerClusterGroup.js

I have to change this everytime I update Leaflet.markercluster.

**Comment by ftechiesnitin** [comment]
Thanks @TomRiedl that did the trick.

**Comment by ghybs** [comment]
Hi,

It seems sad to have to cancel the effect of mobile detection in `_getExpandedVisibleBounds` to prevent this bug.

It is reproducible with Leaflet 1.0.0 and MCG master: http://jsfiddle.net/3v7hd2vx/18/

It is the same issue as #344, which was solved by commit 254198ad47d73ddb7177c6aefde78561046d8c0c.
Unfortunately, the correction was made only in previous master branch (now leaflet-0.7 branch), but not in leaflet-master branch (now master).

Doing the same in current master branch seems to solve the issue: http://jsfiddle.net/3v7hd2vx/19/

Will do a PR shortly.

**Comment by danzel** [comment]
This works with the latest code:
http://jsfiddle.net/3v7hd2vx/192/

**Comment by danzel** [comment]
Ah, I see ghybs fixed it in #644 but it wasn't closed then. Thanks :)
